Automattic/wp-calypso
# Move features data and icons to Automattic data stores

#### Changes proposed in this Pull Request
* Extracted features data to `@automattic/data-store` package and feature icons to `@automattic/onboarding` package. With this change we are making them usable in both Gutenboarding and wp-admin (and anywhere else).
* Added the same redirect condition for Features step as it is currently implemented for Plans step (if no design is selected, then redirect to `/new`). 
  * This fixes a navigation bug in production. You can reproduce it by going to http://wordpress.com/new/features in an incognito window and try to continue to the next step. You'll be redirected to `/new` instead of landing on `/new/plans`.

#### Testing instructions
* Go to `/new` and advance to the `/new/features` step.
* Test feature picker functionality is the same as in production (originally introduced in https://github.com/Automattic/wp-calypso/pull/44431): 
  * by selecting a feature, a "Recommended" badge should  should be displayed  on the highlighted plan at `/new/plans` step.
  * If no feature is selected (step is skipped), `/new/plans` step shows "Popular" badge on the highlighted plan.
  * After creating a site, selected features should be resetted.
* Go to `/new/features?fresh`. You should be redirected to `/new`.

Fixes: https://github.com/Automattic/wp-calypso/issues/45213
